ParkingZone.Data/IRepositories/IRepository.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Microsoft.EntityFrameworkCore.Query.SqlExpressions;
+using ParkingZone.Domain.Commons;
+using System.Linq.Expressions;
+
+namespace ParkingZone.Data.IRepositories;
+
+public interface  IRepository<T> where T : Auditable
```

Comment:
```suggestion
public interface  IRepository<T> where T : class
```

No need in adding not required complexity to the code at this point. Remove Auditable class.